drivers/display/oled/oled.c

```diff
@@ -0,0 +1,320 @@
+	                   0xAA55AA55AA55AA55
+};
+
+/***************************************************************************//**
+ * @brief Turns display on/off.
+ *
+ * @param device - The device structure.
+ *        on_off - DISP_ON
+ *                 DISP_OFF
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t display_on_off(struct ssd_1306_dev *device, uint8_t on_off)
+{
+	int32_t ret = SUCCESS;
+	uint8_t command;
+
+	ret |= gpio_set_value(device->dc_pin, DC_CMD);
+
+	command = (on_off == DISP_ON) ? 0xAF : 0xAE;
+	ret |= spi_write_and_read(device->spi_desc, &command, 1U);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Moves cursor to desired position.
+ *
+ * @param device - The device structure.
+ *        row
+ *        column
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t move_cursor(struct ssd_1306_dev *device, uint8_t row, uint8_t column)
+{
+	int32_t ret = SUCCESS;
+	uint8_t command[3];
+	ret |= gpio_set_value(device->dc_pin, DC_CMD);
+
+	command[0] = 0x21;
+	command[1] = column*8;
+	command[2] = COL_NB * 8 - 1U;
+	ret |= spi_write_and_read(device->spi_desc, &command, 3U);
+
+	command[0] = 0x22;
+	command[1] = row;
+	command[2] = ROWS_NB - 1U;
+	ret |= spi_write_and_read(device->spi_desc, &command, 3U);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Clears data on display.
+ *
+ * @param device - The device structure.
+ *        row    - numbers of rows
+ *        column - numbers of columns
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t display_clear(struct ssd_1306_dev *device, uint8_t rows, uint8_t columns)
+{
+	int32_t ret = SUCCESS;
+	uint8_t i, j;
+
+	ret |= gpio_set_value(device->dc_pin, DC_DATA);
+
+	for(i=0; i<rows; i++)
+		for(j=0; j<columns; j++)
+			ret |= display_print_ascii(device, ' ', i, j);
+
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Prints character at selected position.
+ *
+ * @param device - The device structure.
+ *        row    - row
+ *        column - column
+ *        ascii  - corresponding number to ascii table
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t display_print_ascii(struct ssd_1306_dev *device, uint8_t ascii, uint8_t row, uint8_t column)
+{
+	int32_t ret = SUCCESS;
+	int8_t ch[8];
+	int8_t i;
+
+	ret |= move_cursor(device, row, column);
+
+	ret |= gpio_set_value(device->dc_pin, DC_DATA);
+
+	for(i=0; i<8; i++) //LSB at tx_char[8]
```

Comment:
Is this comment left here on purpose?

Reply:
Also, whitespaces and braces formatting is wrong, go through the linux style guideline first:
https://www.kernel.org/doc/html/v4.10/process/coding-style.html